fbw-a32nx/src/systems/instruments/src/EFB/Failures/FailureGenerators/TakeOffFailureGenerator.tsx

```diff
@@ -0,0 +1,164 @@
+                    100 - settings[genNumber * numberOfSettingsPerGenerator + 4] * 100,
+                    settings[genNumber * numberOfSettingsPerGenerator + 3], 100,
+                    setNewSetting, generatorSettings, genNumber, 3, failureGenContext),
+                FailureGeneratorSingleSetting(t('Failures.Generators.MedSpeedChance'), '%', 0,
+                    100 - settings[genNumber * numberOfSettingsPerGenerator + 3] * 100,
+                    settings[genNumber * numberOfSettingsPerGenerator + 4], 100,
+                    setNewSetting, generatorSettings, genNumber, 4, failureGenContext),
+                FailureGeneratorText(t('Failures.Generators.ClimbingChance'), '%',
+                    chanceClimbing.toString())]}
+        </div>),
+    FailureGeneratorSingleSetting(t('Failures.Generators.MinimumSpeed'), t('Failures.Generators.knots'),
+        0, settings[genNumber * numberOfSettingsPerGenerator + 6],
+        settings[genNumber * numberOfSettingsPerGenerator + 5], 1,
+        setNewSetting, generatorSettings, genNumber, 5, failureGenContext),
+    FailureGeneratorSingleSetting(t('Failures.Generators.SpeedTransLowMed'), t('Failures.Generators.knots'),
+        settings[genNumber * numberOfSettingsPerGenerator + 5],
+        settings[genNumber * numberOfSettingsPerGenerator + 7],
+        settings[genNumber * numberOfSettingsPerGenerator + 6], 1,
+        setNewSetting, generatorSettings, genNumber, 6, failureGenContext),
+    FailureGeneratorSingleSetting(t('Failures.Generators.MaximumSpeed'), t('Failures.Generators.knots'),
+        settings[genNumber * numberOfSettingsPerGenerator + 6], 300,
+        settings[genNumber * numberOfSettingsPerGenerator + 7], 1,
+        setNewSetting, generatorSettings, genNumber, 7, failureGenContext),
+    FailureGeneratorSingleSetting(t('Failures.Generators.MaxHeightAboveRunway'), t('Failures.Generators.feet'), 0, 10000,
+        settings[genNumber * numberOfSettingsPerGenerator + 8], 100,
+        setNewSetting, generatorSettings, genNumber, 8, failureGenContext)];
+    return settingTable;
+};
+
+export const failureGeneratorTakeOff = (generatorFailuresGetters : Map<number, string>) => {
+    const [absoluteTime500ms] = useSimVar('E:ABSOLUTE TIME', 'seconds', 500);
+    const { maxFailuresAtOnce, totalActiveFailures, allFailures, activate, activeFailures } = failureGeneratorCommonFunction();
+    const [failureGeneratorSetting, setFailureGeneratorSetting] = usePersistentProperty(settingName, '');
+
+    const settings : number[] = useMemo<number[]>(() => failureGeneratorSetting.split(',').map(((it) => parseFloat(it))), [failureGeneratorSetting]);
+
+    const nbGenerator = useMemo(() => Math.floor(settings.length / numberOfSettingsPerGenerator), [settings]);
+    const { failureFlightPhase } = basicData();
+
+    const altitude = SimVar.GetSimVarValue('PLANE ALTITUDE', 'feet');
+    const gs = SimVar.GetSimVarValue('GPS GROUND SPEED', 'knots');
```

Comment:
Hmm is there a reason this is using ``SimVar.GetSimVarValue`` ?

Reply:
Can you elaborate your concern ?
Is this to be avoided ?
How would I retrieve these data otherwise, any other option ?

Reply:
Usually we’re using the useSimvar hooks to retrieve simvars

Reply:
Thanks @Saschl I will look into replacing it.

Reply:
Replaced in the code